src/sixel/mod.rs

```diff
@@ -0,0 +1,172 @@
+  let mut app = App::default();
+  let mut set_selected = false;
+  for palette in &mut app.fractal.palette {
+    palette.interpolation = InterpolationMode::None;
+  }
+
+  while !set_selected {
+    clear_terminal();
+    print!("Choose a fractal set to render:\n1. Mandelbrot\n2. Julia\n3. Burning Ship\n4. Quit\n>");
+    io::stdout().flush().unwrap();
+    let mut input = String::new();
+    if io::stdin().read_line(&mut input).is_err() {
+      continue;
+    }
+
+    match input.trim() {
+      "1" => {
+        app.fractal.set = Set::Mandelbrot;
+        set_selected = true;
+      }
+      "2" => {
+        app.fractal.set = Set::Julia;
+        set_selected = true;
+      }
+      "3" => {
+        app.fractal.set = Set::BurningShip;
+        set_selected = true;
+      }
+      "4" => exit(0),
+      _ => continue,
+    }
+    clear_terminal();
+  }
+
+  let mut command_mode = false;
+  let mut command_string = String::new();
+
+  enable_raw_mode().unwrap();
+  let mut first_render = true;
+  let mut refresh = false;
+
+  loop {
+    let mut last_key = None;
+    while event::poll(Duration::from_millis(0)).unwrap() {
+      if let Event::Key(key) = event::read().unwrap() {
+        last_key = Some(key);
+      }
+    }
+
+    if let Some(key) = last_key {
+      refresh = true;
+      if !command_mode {
+        match key.code {
+          KeyCode::Char('q') => break,
+          KeyCode::Char('w') => app.fractal.z.im -= 0.1 / app.fractal.scale,
+          KeyCode::Char('s') => app.fractal.z.im += 0.1 / app.fractal.scale,
+          KeyCode::Char('a') => app.fractal.z.re -= 0.1 / app.fractal.scale,
+          KeyCode::Char('d') => app.fractal.z.re += 0.1 / app.fractal.scale,
+          KeyCode::Char('=') => app.fractal.scale *= 1.1,
+          KeyCode::Char('-') => app.fractal.scale /= 1.1,
+          KeyCode::Char('r') => app.fractal.max_iterations += 1,
+          KeyCode::Char('f') => app.fractal.max_iterations -= 1,
+          KeyCode::Char(':') => {
+            command_mode = true;
+            command_string.clear();
+          }
+          _ => refresh = false,
+        }
+      } else {
+        match key.code {
+          KeyCode::Esc => {
+            command_mode = false;
+            command_string.clear();
+          }
+          KeyCode::Enter => {
+            let cmd = parse_command(&command_string);
+            let _ = execute_command(&mut app, cmd).unwrap_or_else(|err| format!("Error: {}", err));
+            command_mode = false;
+            command_string.clear();
+          }
+          KeyCode::Backspace => {
+            command_string.pop();
+          }
+          KeyCode::Char(c) => {
+            command_string.push(c);
+          }
+          _ => {}
+        }
+      }
+    }
+
+    if first_render || refresh {
+      first_render = false;
+      refresh = false;
+      let img = app.fractal.render_frame(WIDTH, HEIGHT, false);
+      let mut out = String::new();
+
+      out.push_str("\x1bPq");
```

Comment:
You may want to try `"\x1bP9;1q"` here. The `9` just specifies a 1:1 aspect ratio, which you're already doing later, but the important part is the second parameter - the `1` - which tells the terminal that it should _not_ erase the background before drawing the image. That might help with the flickering.

Reply:
That suggests the "black" parts are not actually black but transparent. So while it may appear to work correctly when you let the terminal erase the background, you should be aware that the erase color is not guaranteed to be black. It would be preferable if you could force those transparent pixels to map to the black palette entry if that's the color they're expected to show.